resolwe/flow/views/data.py

```diff
@@ -180,8 +187,22 @@ def parents(self, request, pk=None):
         return self._parents_children(request, self.get_object().parents)
 
     @extend_schema(
-        filters=False, responses={status.HTTP_200_OK: DataSerializer(many=True)}
+        filters=False,
+        request=DuplicateDataSerializer(),
+        responses={status.HTTP_200_OK: DataSerializer(many=True)},
     )
+    @action(detail=True, methods=["post"])
+    def restart(self, request, pk=None):
+        """Restart the current data object."""
+        argument_validator = RestartSerializer(data=request.data)
+        argument_validator.is_valid(raise_exception=True)
+        data = self.get_object()
+        try:
+            data.restart(**argument_validator.validated_data)
```

Comment:
Only admin users should be able to restart the process.

Reply:
Done.